src/main/java/org/opensearch/ad/rest/RestCreateDetectorAction.java

```diff
@@ -1,134 +1,227 @@
     public List<Route> routes() {
-        return List.of(new Route(GET, "/hello"), new Route(PUT, "/hello/{name}"));
+        return List.of(new Route(POST, "/detectors"));
     }
 
-    @Override
-    public ExtensionRestResponse handleRequest(ExtensionRestRequest request) {
-        Method method = request.method();
+    private String getAnomalyDetectorMappings() throws IOException {
+        URL url = AnomalyDetectionIndices.class.getClassLoader().getResource(ANOMALY_DETECTORS_INDEX_MAPPING_FILE);
+        return Resources.toString(url, Charsets.UTF_8);
+    }
+
+    private IndexResponse indexAnomalyDetector(AnomalyDetector anomalyDetector) throws IOException {
+        AnomalyDetector detector = new AnomalyDetector(
+            anomalyDetector.getName(),
+            anomalyDetector.getVersion(),
+            anomalyDetector.getName(),
+            anomalyDetector.getDescription(),
+            anomalyDetector.getTimeField(),
+            anomalyDetector.getIndices(),
+            anomalyDetector.getFeatureAttributes(),
+            anomalyDetector.getFilterQuery(),
+            anomalyDetector.getDetectionInterval(),
+            anomalyDetector.getWindowDelay(),
+            anomalyDetector.getShingleSize(),
+            anomalyDetector.getUiMetadata(),
+            anomalyDetector.getSchemaVersion(),
+            Instant.now(),
+            anomalyDetector.getCategoryField(),
+            null,
+            anomalyDetector.getResultIndex()
+        );
+
+        IndexRequest<AnomalyDetector> indexRequest = new IndexRequest.Builder<AnomalyDetector>()
+            .index(ANOMALY_DETECTORS_INDEX)
+            .id("1")
+            .document(detector)
+            .build();
+        IndexResponse indexResponse = sdkClient.index(indexRequest);
+        return indexResponse;
 
-        if (Method.GET.equals(method)) {
-            return handleGetRequest(request);
-        } else if (Method.POST.equals(method)) {
-            return handlePostRequest(request);
-        } else if (Method.DELETE.equals(method)) {
-            return handleDeleteRequest(request);
-        } else if (Method.PUT.equals(method)) {
-            return handlePutRequest(request);
-        }
-        return handleBadRequest(request);
     }
 
-    private ExtensionRestResponse handleGetRequest(ExtensionRestRequest request) {
-        String worldNameWithRandomAdjective = worldAdjectives.isEmpty()
-            ? worldName
-            : String.join(" ", worldAdjectives.get(rand.nextInt(worldAdjectives.size())), worldName);
-        return new ExtensionRestResponse(request, OK, String.format(GREETING, worldNameWithRandomAdjective));
+    public List<NamedXContentRegistry.Entry> getNamedXWriteables() {
+        List<NamedXContentRegistry.Entry> entries = new ArrayList<>();
+        entries.add(AnomalyDetector.XCONTENT_REGISTRY);
+        entries.add(AnomalyResult.XCONTENT_REGISTRY);
+        entries.add(DetectorInternalState.XCONTENT_REGISTRY);
+        entries
```

Comment:
Note to other reviewers: these four `entries.add()` here and below are placeholder code and will be replaced by getNamedXContent work in progress.

This comment may be resolved when ready to merge.